rewrite-maven/src/main/java/org/openrewrite/maven/AddOrUpdateDependencyManagement.java

```diff
@@ -0,0 +1,291 @@
+            p = p.withDependencyManagement(p.getDependencyManagement().withDependencies(dependencies));
+            return p;
+        }
+    }
+
+    private class UpdateDependency extends MavenRefactorVisitor {
+        @Override
+        public Maven visitPom(Maven.Pom pom) {
+            Maven.Pom p = refactor(pom, super::visitPom);
+            List<Maven.Dependency> dependencies = new ArrayList<>(pom.getDependencyManagement().getDependencies());
+            Maven.Dependency dependency = findDependencyWithGroupIdAndArtifactId(dependencies, groupId, artifactId);
+
+            final Xml.Tag tag = createDependencyTag(formatter.wholeSourceIndent());
+            final Maven.Dependency updatedDependency = new Maven.DependencyManagement.Dependency(false,
+                    new MavenModel.Dependency(
+                            new MavenModel.ModuleVersionId(groupId, artifactId, null, version, "jar"),
+                            version,
+                            scope
+                    ),
+                    tag
+            );
+
+            dependencies.set(dependencies.indexOf(dependency), updatedDependency);
+            p = p.withDependencyManagement(p.getDependencyManagement().withDependencies(dependencies));
+            return p;
+        }
+
+        private Maven.Dependency findDependencyWithGroupIdAndArtifactId(List<Maven.Dependency> dependencies, String groupId, String artifactId) {
+            return dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst().get();
+        }
+
+        private int getIndexOfDependencyToBeRemoved(List<Maven.Dependency> dependencies) {
+            final Optional<Maven.Dependency> optionalDependency = dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst();
+            return dependencies.indexOf(optionalDependency.get());
+        }
+
+        private void removeDependency(List<Maven.Dependency> dependencies) {
+            final Optional<Maven.Dependency> optionalDependency = dependencies.stream()
+                    .filter(d -> groupId.equals(d.getGroupId()) && artifactId.equals(d.getArtifactId()))
+                    .findFirst();
+            final Maven.Dependency dependency = optionalDependency.get();
+            dependencies.remove(dependency);
+        }
+    }
+
+    private class AddOrUpdateDependency extends MavenRefactorVisitor {
+        @Override
+        public Maven visitPom(Maven.Pom pom) {
+            if (dependencyWithGroupIdAndArtifactIdExists(pom)) {
+                andThen(new AddDependenciesTagIfNotPresent());
+                andThen(new UpdateDependency());
+            } else if (dependencyDoesNotExist(pom)) {
+                andThen(new AddDependenciesTagIfNotPresent());
+                andThen(new InsertDependencyInOrder());
+            }
```

Comment:
Minor nitpick: It's slightly inefficient/redundant to essentially call `dependencyWithGroupIdAndArtifactExists()` twice.
What do you think about this suggestion?

```suggestion
            if(pom.getDependencyManagement() != null) {
                if (dependencyWithGroupIdAndArtifactIdExists(pom)) {
                    andThen(new AddDependenciesTagIfNotPresent());
                    andThen(new UpdateDependency());
                } else {
                    andThen(new AddDependenciesTagIfNotPresent());
                    andThen(new InsertDependencyInOrder());
                }
            }
```

Then `dependencyDoesNotExist()` could also be deleted